### PR TITLE
feat(dataset): wildcard support when adding data from git

### PR DESCRIPTION
**Pull request by mohammad-sdsc**
# Description

Now, you can use wildcards when selecting which files to be added from a git repo:
```
renku dataset add my-dataset -s notebooks/**/*py -s notebooks/*renku https://github.com/SwissDataScienceCenter/r10e-ds-py.git
```

### CODE REVIEW

**Comment by rokroskar** [comment]
Will this wildcard repeat the selection when you run `renku dataset update`?

**Comment by mohammad-sdsc** [comment]
> Will this wildcard repeat the selection when you run `renku dataset update`?

No, it won't. This is evaluated at the time the command is executed and gets translated to a list of files.

**Comment by rokroskar** [comment]
So should the workflow be then to just repeat the `add` command with the wildcard? will it fail if the files are already there?

**Comment by mohammad-sdsc** [comment]
> So should the workflow be then to just repeat the `add` command with the wildcard? will it fail if the files are already there?

True. Repeating the workflow would work but we need to pass `--force` flag too. Adding of existing files fails unless this flag is passed in.

**Comment by rokroskar** [comment]
would it make sense to have a second one, e.g. `--ignore-existing` or similar?

File: renku/cli/dataset.py [hunk 1]
```diff
@@ -118,6 +118,14 @@
       my-dataset/
         datafile
 
+You can use shell-like waldcards (e.g. *, **, ?) when specifying paths to be
+added:
+
+.. code-block:: console
+
+    $ renku dataset add my-dataset --source path/**/datafile \
```
Comment (rokroskar) [inline renku/cli/dataset.py]: do the options with asterisks need to be in quotes?

Comment (mohammad-sdsc) [reply renku/cli/dataset.py]: It's better to do it. If the path does not exist in the CWD then shell won't expand it and it's safe to use it without quotes.
I'll add quotes in the example.

**Pull request merged**